packages/astro/src/core/render/result.ts

```diff
@@ -150,10 +144,10 @@ export function createResult(args: CreateResultArgs): SSRResult {
 
 			const Astro = {
 				__proto__: astroGlobal,
-				canonicalURL,
 				params,
 				props,
 				request,
+				url: new URL(request.url),
```

Comment:
Wasn't this already created up here: https://github.com/withastro/astro/pull/3959/files#diff-8324bd7b157e6b5bd22f2e30606dbf85d7f53b0be9449df7210c2b5e1003ebc5R110

Reply:
yup, missed that! Adding it now.